### DISCUSSION TITLE
Howto - installation method similar to setup.py?

### DISCUSSION THREAD

**Discussion by edrozenberg** [fastapi/fastapi#8598] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
python3 setup.py install --root=/some/dir/
```

### Description

Would like to package up fastapi by using an installation similar to `python3 setup.py install --root=$PKG`

fastapi does not provide a `setup.py`. As a packager/admin, I do not plan to use `pip` for the installation - rather I plan to create a package and push it out to a number of servers.

Documentation does not describe any method other than pip. I don't use pip to install most things, I package up every python item individually.

### Operating System

Linux

### Operating System Details

Slackware

### FastAPI Version

0.73.0

### Python Version

Python 3.9.10

**Comment by edrozenberg** [comment]
For now I'm exploding (aka unzipping) the wheel from https://pypi.org/project/fastapi/#files and copying into the pkg manually.

**Comment by jgould22** [comment]
Fastapi appears to use `flit` for building/publishing

It appears flit can produce a setup.py if that is what you want - https://flit.readthedocs.io/en/latest/cmdline.html#flit-build

**Comment by Kludex** [comment, marked as answer]
`pip install -e .` should work nowadays.
